src/mpid/ch4/shm/ipc/xpmem/xpmem_mem.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+#ifndef XPMEM_MEM_H_INCLUDED
+#define XPMEM_MEM_H_INCLUDED
+
+#include "ch4_impl.h"
+#include "xpmem_pre.h"
+#include "xpmem_seg.h"
+
+MPL_STATIC_INLINE_PREFIX int MPIDI_IPC_xpmem_get_mem_handle(void *vaddr, size_t size,
+                                                            MPIDI_IPC_xpmem_mem_handle_t *
+                                                            mem_handle)
+{
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_IPC_XPMEM_GET_MEM_HANDLE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IPC_XPMEM_GET_MEM_HANDLE);
+
+    mem_handle->src_offset = (uint64_t) vaddr;
+
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_IPC_XPMEM_GET_MEM_HANDLE);
+    return MPI_SUCCESS;
+}
```

Comment:
Do we really need these to be static inlines?  These are only used for large messages.

Reply:
Sorry I missed this one. These function will all be deleted in favor of a generic implementation of AVL tree. I agree that for large messages the benefit of making them inlined is probably minimal. If you really want I can move them to `c` file.

Reply:
Yes, please do.  Excessive inlining hurts the build time.

Reply:
fixed